### PR TITLE
CLDR-14251 Modernize: continue implementing front end without dojo

### PR DESCRIPTION
**Pull request by btangmu**
-Fix bug in PR #893: in cldrAjax.js, only use responseText if responseType is text

-Section view is now mostly working without dojo

-Add lots of placeholders for old dojo-dependent features

-New cldrEvent.js has most of the code from old redesign.js

-Move OtherSpecial to new cldrOtherSpecial.js

-Remove dead code; clean up

-Comments

##### Checklist

- [x] Issue filed: https://unicode-org.atlassian.net/browse/CLDR-14251
- [x] Updated PR title and link in previous line to include Issue number

### CODE REVIEW

File: tools/cldr-apps/src/main/webapp/AdminPanel.jsp
Comment (srl295) [inline tools/cldr-apps/src/main/webapp/AdminPanel.jsp:136]: 👍

File: tools/cldr-apps/src/main/webapp/js/new/cldrLoad.js
Comment (srl295) [inline tools/cldr-apps/src/main/webapp/js/new/cldrLoad.js:1739]: nice. I'll look forward to seeing this.


Any thought of splitting out `specialItems[]` to a separate file? I know it has code in it, but that could be abstracted as an input.

File: tools/cldr-apps/src/main/webapp/js/new/cldrOtherSpecial.js
Comment (srl295) [inline tools/cldr-apps/src/main/webapp/js/new/cldrOtherSpecial.js:2]: '…possibly a class' - +1 on that, is that being tracked in any of the issues?

File: tools/cldr-apps/src/main/webapp/js/new/cldrSurvey.js
Comment (srl295) [inline tools/cldr-apps/src/main/webapp/js/new/cldrSurvey.js:21]: it should be a singleton.

File: tools/cldr-apps/src/main/webapp/js/new/cldrSurvey.js
Comment (srl295) [inline tools/cldr-apps/src/main/webapp/js/new/cldrSurvey.js:59]: good candidate for splitting out as static data

File: tools/cldr-apps/src/main/webapp/js/new/cldrSurvey.js
Comment (srl295) [inline tools/cldr-apps/src/main/webapp/js/new/cldrSurvey.js:4273]: FWIW you can write this as:

```js
return {
 INHERITANCE_MARKER,
 addClass,
 addIcon,
…
}
```

If the variable and the hash key have the exact same name, the variable doesn't have to be repeated.

**Comment by btangmu** [comment]
FYI, there are lots of further changes in https://github.com/unicode-org/cldr/pull/899

I'm not sure if it's better to merge this one, or that one after more testing. Either way, comments are welcome!

**Review by srl295 — COMMENTED** [review]
> FYI, there are lots of further changes in #899
> 
> I'm not sure if it's better to merge this one, or that one after more testing. Either way, comments are welcome!

hiya! I'm reviewing 898 right now, maybe it would be worth merging this one first if it gets reviewed?

**Review by srl295 — APPROVED** [review]
looks great to me!

**Comment by srl295** [comment]
> FYI, there are lots of further changes in #899
> 
> I'm not sure if it's better to merge this one, or that one after more testing. Either way, comments are welcome!

You could rebase #899 on top of `btangmu:t14251_g` — that would make #899 have 2 commits, but then the delta could be reviewed separately

**Comment by srl295** [comment]
@btangmu
your dilemma about multiple PRs kind of shows a challenge about the way we handle feature branches and "one commit per PR" with these kind of changes…

I wonder if, given the size and scope of this effort, and the very low probability that someone else is going to commit on top of your files, if it might be better to have a long running feature branch on the head fork, <https://github.com/unicode-org/cldr/tree/btangmu-newsurvey> or something.   Then, you could put multiple  commits directly on that branch and I could review the commits individually, or in bulk… but basically, free you from needing to squash everything until the very end.  And then make a larger squashed PR merging that branch in.

If it's easier to just proceed as you are then go ahead…

File: tools/cldr-apps/src/main/webapp/js/new/cldrLoad.js
Comment (btangmu) [reply tools/cldr-apps/src/main/webapp/js/new/cldrLoad.js:1739]: I'm thinking of moving most menu-related code from cldrLoad.js into a new cldrMenu.js, then sorting out the dependencies. It also sounds like a good idea to further separate between code and data. Also, the gear menu maybe shouldn't act like a pull-down menu; clicking the gear icon could simply open a page with "#gear" in the url.

File: tools/cldr-apps/src/main/webapp/js/new/cldrOtherSpecial.js
Comment (btangmu) [reply tools/cldr-apps/src/main/webapp/js/new/cldrOtherSpecial.js:2]: I'll make sure this TODO is dealt with somehow before this ticket is closed; maybe a new ticket. So far, cldrOtherSpecial.js is just a placeholder. I'm re-implementing what were "specials", such as `#forum`, one-by-one, and also adding some new "specials" like `#about` and `#admin`, but without using the js/specials folder or dojo. I'm still studying how to capture what the "specials" all share in common in a more object-oriented way.

File: tools/cldr-apps/src/main/webapp/js/new/cldrSurvey.js
Comment (btangmu) [reply tools/cldr-apps/src/main/webapp/js/new/cldrSurvey.js:21]: Thanks! BTW the way it's initialized here assumes cldrXpathMap.js is loaded before cldrSurvey.js. PR 899 uses lazy initialization instead, with a getter. I'm testing by loading the js files in alphabetical order, and the reverse order, trying to make sure the order doesn't matter. Soon we should be ready for some tools like Webpack to help fix such problems.

File: tools/cldr-apps/src/main/webapp/js/new/cldrSurvey.js
Comment (btangmu) [reply tools/cldr-apps/src/main/webapp/js/new/cldrSurvey.js:59]: yes

File: tools/cldr-apps/src/main/webapp/js/new/cldrSurvey.js
Comment (btangmu) [reply tools/cldr-apps/src/main/webapp/js/new/cldrSurvey.js:4273]: I didn't know that. Very cool improvement in ES6! The IIFE will be so much less clunky without the duplication. I've been picturing that we'll use "export" instead of IIFE, but who knows maybe some combination...

**Comment by btangmu** [comment]
Thanks! I'll consider your ideas about a long-running feature branch. Since there's no submission/vetting for v39 we have more flexibility. Thanks for reviewing this in the middle of the holiday break! Happy New Year!!

**Comment by btangmu** [comment]
> > FYI, there are lots of further changes in #899
> > I'm not sure if it's better to merge this one, or that one after more testing. Either way, comments are welcome!
> 
> You could rebase #899 on top of `btangmu:t14251_g` — that would make #899 have 2 commits, but then the delta could be reviewed separately
> 
> **edit** a lot of merge conflicts that way. though.. hmm

Here's the recipe I used after merging this PR; my most recent changes were in branch t14251_k:

git fetch upstream
git checkout master
git reset --hard upstream/master
git push origin master --force
git checkout -b t14251_m
git merge --squash -X theirs t14251_k

Result: "Automatic merge went well" -- no conflicts had to be merged manually :-)

**Comment by srl295** [comment]
🎉 good way to end the year

**Pull request merged**
